app/services/subscriptions/terminate_service.rb

```diff
@@ -70,7 +70,7 @@ def terminate_and_start_next(timestamp:)
       else
         [subscription]
       end
-      BillSubscriptionJob.perform_later(billable_subscriptions, timestamp)
+      BillSubscriptionJob.perform_later(billable_subscriptions, timestamp, invoice_type: :upgrading)
```

Comment:
Here billable_subscriptions typically has 2 subscriptions. The one you just ended and the new one you upgraded too. I'd expect 2 different types. The old one would be `terminating` and the new one would be `starting`.

If there is only one type per group of subscription, then maybe the type should be on Invoice rather than InvoiceSubscription 🤔

Reply:
Actually an invoice can be generated for:
- A single starting subscription
- A single terminating subscription
- One or multiple periodic invoice subscriptions
- A single terminating subscription and a starting subscription.

What we want to enforce:
- No duplication for the same boundaries on periodic invoices (done using the recurring flag)
- No duplication for the start of a single subscription
- No duplication for the termination of a single subscription
Those two lasts are the ones targeted by this PR.

The flag/type cannot be at invoice level, because of the n-n relation between subscriptions and invoices.

Let imagine the following case:
- A subscription upgraded
- As a consequence the subscription is terminated and a new one is started
- An invoice is generated with the fees for the terminated one and the subscription fees to the new one
- Because of some race condition a terminating invoice is emitted for the initial invoice 
  -> it should fail, but today it's not and we end-up with two invoices with the same boundaries